### PR TITLE
Fixed inconsistency of `val` vs `value` in OM

### LINKED ISSUE #1834: Inconsistent application of Val vs. Value
### Summary of Issue

Inconsistent use of `val` vs. `value` 

### Issue Type

- [X] Bug
- [ ] Enhancement  
      POEM #N/A

### Description

In `ExecComp()` initial values are specified using `value=` 
However in component definitions, initial conditions are specified using `val=`

Should have a single nomenclature for setting values in all of OpenMDAO.

### Example

ExecComp() Example:

```
p.model.add_subsystem('LengthCable', om.ExecComp(['length_cable = 2*((L_arm+D_sphere)*4+L_Aux+L_TMS)'],
                                               length_cable = {'value':0, 'units':'m'},
                                               L_arm={'value':0, 'units':'m'}, 
                                               L_Aux={'value':0, 'units':'m'},
                                               L_TMS={'value':0, 'units':'m'},
                                               D_sphere={'value':0, 'units':'m'}),
                  promotes=['*'])
```

Component Definition Example:
```
class mach(ExplicitComponent):

    def setup(self):

        # Inputs
        self.add_input('V_inf', val=20, desc='free stream velocity', units='m/s')
        
        # Outputs
        self.add_output('Mach', val=np.zeros(nn), desc='mach number', units=None)
```
        
### Environment

N/A

### PR DESCRIPTION
**Pull request by DKilkenny**
### Summary

Fixed inconsistency of `val` vs `value` in OM

### Related Issues

- Resolves #1834

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

**Review by naylor-b — APPROVED** [review]

File: openmdao/components/exec_comp.py [hunk 1]
```diff
@@ -339,6 +341,16 @@ def _setup_expressions(self):
                                        "recognized for variable '%s': %s" %
                                        (self.msginfo, arg, sorted(diff)))
 
+                if 'val' in val and 'value' in val:
+                    raise RuntimeError("Cannot use 'val' and 'value' at the same time, use 'val'.")
+                elif 'value' in val and not warned:
+                    warn_deprecation("'value' will be deprecated in 4.0. Please use 'val' in the "
```
Comment (swryan) [inline openmdao/components/exec_comp.py:347]: I think you need `self.msginfo` in this warning message to point the user to the component in question.
And I think you mean to say that the argument *is* deprecated and will be *removed* in 4.0?
Also, I think you could clean this logic up a bit by merging with the following if statement, e.g.:
```
elif 'value' in val:
    if not warned:
        warn_deprecation(f"{self.msginfo}: the 'value' argument is deprecated and will be removed in version 4.0. Please use 'val'.")
        warned = True
    val['val'] = val.pop('value')
```

**Review by Kenneth-T-Moore — APPROVED** [review]

File: openmdao/components/exec_comp.py [hunk 2]
```diff
@@ -424,7 +436,10 @@ def _setup_expressions(self):
                     dct = self.add_input(var, val, **meta)
 
             if var not in init_vals:
-                init_vals[var] = dct['value']
+                if 'value' in dct:
+                    dct['val'] = dct.pop('value')
```
Comment (swryan) [inline openmdao/components/exec_comp.py:440]: not clear that this is needed?  `value` has been replaced by `val` in `kwargs2`/`meta` right?

File: openmdao/components/exec_comp.py [hunk 3]
```diff
@@ -339,6 +341,16 @@ def _setup_expressions(self):
                                        "recognized for variable '%s': %s" %
                                        (self.msginfo, arg, sorted(diff)))
 
+                if 'val' in val and 'value' in val:
+                    raise RuntimeError("Cannot use 'val' and 'value' at the same time, use 'val'.")
```
Comment (swryan) [inline openmdao/components/exec_comp.py:345]: I think you need `self.msginfo` in this warning message to point the user to the component in question.

File: openmdao/core/problem.py [hunk 4]
```diff
@@ -408,23 +408,31 @@ def __setitem__(self, name, value):
...
-            Indices or slice to set to specified value.
+            Indices or slice to set to specified val.
+        value : float or ndarray or list or None
+            This argument is deprecated. Use 'val' instead.
         """
+        if val is None and value is None:
+            raise RuntimeError("Val is required.")
+        elif value is not None:
+            val = value
+            warn_deprecation("'value' will be deprecated in 4.0. Please use 'val' in the future.")
```
Comment (swryan) [inline openmdao/core/problem.py:434]: I think you need some additional info in this warning message to point the user to the call in question
And I think you mean to say that the argument *is* deprecated and will be *removed* in 4.0?
e.g.:
```
warn_deprecation(f"{self.msginfo} when calling `set_val` on {name}, the 'value' argument is deprecated and will be removed in version 4.0. Please use 'val'.")
```

**Review by swryan — CHANGES_REQUESTED** [review]

**Review by robfalck — APPROVED** [review]

File: openmdao/core/system.py [hunk 5]
```diff
@@ -3511,7 +3511,7 @@ def get_io_metadata(self, iotypes=('input', 'output'), metadata_keys=None,
         return result
 
     def list_inputs(self,
-                    values=True,
+                    vals=True,
```
Comment (robfalck) [inline openmdao/core/system.py:3514]: I think this should be singular `val` to match `shape`, as well as the use of `val` everywhere else in the code.

File: openmdao/core/system.py [hunk 6]
```diff
@@ -3616,7 +3633,7 @@ def list_inputs(self,
 
     def list_outputs(self,
                      explicit=True, implicit=True,
-                     values=True,
+                     vals=True,
```
Comment (robfalck) [inline openmdao/core/system.py:3636]: Singular `val` instead of `vals` here also.

File: openmdao/recorders/case.py [hunk 7]
```diff
@@ -392,7 +393,7 @@ def get_responses(self, scaled=True, use_indices=True):
         return self._get_variables_of_type('response', scaled, use_indices)
 
     def list_inputs(self,
-                    values=True,
+                    vals=True,
```
Comment (robfalck) [inline openmdao/recorders/case.py:396]: `val` instead of `vals`

File: openmdao/recorders/case.py [hunk 8]
```diff
@@ -488,7 +501,7 @@ def list_inputs(self,
 
     def list_outputs(self,
                      explicit=True, implicit=True,
-                     values=True,
+                     vals=True,
```
Comment (robfalck) [inline openmdao/recorders/case.py:504]: `val` instead of `vals`

**Review by robfalck — CHANGES_REQUESTED** [review]

File: openmdao/recorders/case.py (hunk 8 above)
Comment (DKilkenny) [reply openmdao/recorders/case.py]: Fixed

**Review by robfalck — APPROVED** [review]

**Review by swryan — APPROVED** [review]

**Pull request merged**